implement VisitFuncCallNode for function object

Kouhei-Moriya opened the pull request:
change to do not use ZFuncType.TypeParams directly
report type error in VisitNode
change object structure

Kouhei-Moriya committed:
change to do not use ZFuncType.TypeParams directly
change to report type error
change object structure
implement VisitFuncCallNode for function object

imasahiro commented on src/zen/codegen/llvm/LLVMSourceGenerator.java:
```diff
@@ -224,6 +224,9 @@ public String Write() {
 
+	private static final boolean WithInitValue = true;
+	private static final boolean WithoutInitValue = false;
+
```
Please merge "WithInitValue" and "WithoutInitValue".

Kouhei-Moriya commented:
Which expression is correct, "private static final boolean WithoutInitValue = !WithInitValue;" or "this.VisitFieldList("i8*", Node, !WithInitValue);"(in line 666) ?

imasahiro commented:
The latter is better.

Kouhei-Moriya committed:
merge WithInitValue and WithoutInitValue

Kouhei-Moriya committed:
support bit operator & | << >> ^ ~
add method (Declare|Call)ExternalFunction, GetSizeOfType
implement VisitLetNode and VisitComparatorNode for object
use external (malloc|memcpy) in VisitNewObjectNode
rename constructor symbol and call constructor in VisitNewObjectNode
